// @crossmarkio/typings/build/src/schema/v3/api/general.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeneralSchema = void 0;
const v2_1 = require("../../v2");
const themes_1 = require("../themes");
const v2 = v2_1.Types.Api.Types;
class GeneralSchema {
    version;
    initialized = false;
    language = 'en';
    discreetMode = false;
    hideZero = false;
    hideNegative = true;
    popupMode = false;
    theme = themes_1.Themes.availableThemes.theme1;
    active;
    static migration = (oldValue, activeCardId) => {
        const Switch = {
            light: themes_1.Themes.availableThemes.theme1,
            dark: themes_1.Themes.availableThemes.theme2,
            sun: themes_1.Themes.availableThemes.theme3,
            moon: themes_1.Themes.availableThemes.theme4,
        };
        let newTheme = themes_1.Themes.availableThemes.theme1;
        if (oldValue.theme)
            newTheme = Switch[oldValue.theme] || newTheme;
        return Object.assign({}, oldValue, {
            theme: newTheme,
            active: activeCardId,
        });
    };
}
exports.GeneralSchema = GeneralSchema;
